plug.py

```diff
@@ -312,6 +312,31 @@ def notify_newdata(data, is_beta):
     return error
 
 
+def notify_journal_entry_cqc(cmdr, is_beta, entry, state):
+    """
+    Send a journal entry to each plugin.
+    :param cmdr: The Cmdr name, or None if not yet known
+    :param entry: The journal entry as a dictionary
+    :param state: A dictionary containing info about the Cmdr, current ship and cargo
+    :param is_beta: whether the player is in a Beta universe.
+    :returns: Error message from the first plugin that returns one (if any)
+    """
+
+    error = None
+    for plugin in PLUGINS:
+        cqc_callback = plugin._get_func('journal_entry_cqc')
+        if cqc_callback is not None and callable(cqc_callback):
+            try:
+                # Pass a copy of the journal entry in case the callee modifies it
+                newerror = cqc_callback(cmdr, is_beta, dict(entry), dict(state))
```

Comment:
dont use `dict()`, that creates a shallow copy, use `copy.deepcopy` if you want to be sure